src/app/chat/chat.service.ts

```diff
@@ -128,29 +160,39 @@ export class ChatService {
    *  size:20
    * }
    */
-  getMessageList(prams: messageListPrams): Observable<any> {
-    console.log("Service", this.messageList);
-    return of(this.messageList);
+  getMessageList(data: messageListPrams): Observable<any> {
+    return this.request.get(api.getChatMessages, {
+      params: data
+    });
   }
 
   markMessagesAsSeen(prams: markAsSeenPrams): Observable<any> {
-    prams.action = 'mark_seen';
-    return of("maked");
+    let body = {
+      'team_id': prams.team_id,
+      'id': prams.id,
+      'action': 'mark_seen'
+    }
+    return this.request.post(api.markAsSeen, body, {
+      headers: { 'Content-Type': 'application/x-www-form-urlencoded' }
+    });
   }
 
   postNewMessage(data: newMessage): Observable<any> {
-    data.env = 'develop';
-    let returnData = {
-      id: 300,
-      sender_name: 'Chathumal',
-      is_sender: true,
-      team_name: 'dream team',
-      sent_time: '1.30 PM',
-      message: data.message,
-      team_id: data.team_id,
-      to: data.to
-    };
-    return of(returnData);
+    const body = new HttpParams()
```

Comment:
Don't need to create a HttpParams. We can just pass an object of body parameters to `this.request.post()`

Reply:
for this api request we need to pass those data in body as Form data.. if we didn't pass it not working. 
i try to pass data as params and pass empty object to request body. but didn't work. this is also happening in login.